### DISCUSSION TITLE
Missing eslint plugin:react-hooks/recommended

### DISCUSSION THREAD

**Discussion by guiccbr** [infinitered/ignite#2453] [Q&A]
Hello, everyone, I just started a new app with Ignite but then noticed that the eslint plugin `react-hooks/recommended` is not enabled, meaning that rules-of-hooks will not be linted by default.

Since enabling these rules is [strongly recommended in the react docs](https://react.dev/learn/editor-setup#linting), I'm curious to understand if there is any reason for not having it configured in this template, or if the lack of the plugin was unintentional.

**Comment by joshuayoes** [comment]
H guiccbr, thanks for opening this discussion!

In my experience, it seems that Infinite Red has leaned away from excessive linting rules in client and open source projects. However, `react-hooks/recommended` is a useful rule set in my opinion. 

Would you be open to adding a PR to add this?

**Reply by guiccbr** [reply]
Sure, I can open a PR for that.

I do have a question though. It's not clear to me if adding that eslint rule set could potentially cause trouble with users of this library that already have a project set up. I don't believe so, because I believe that after you set up, you should not need to ever "upgrade" your setup to use a newer version of ignite right? Is that true?

**Reply by joshuayoes** [reply]
As Ignite currently is, you cannot update to future versions. The boilerplate is a starting point from the version that it is created, and then future changes are forked from there. So you would be safe to add the eslint rule set and not break future ignite apps.

**Reply by joshuayoes** [reply]
Essentially, ignite is like git cloning a template, but it has a few nice features like renaming all the references to your app name to get it started on top of it.

**Reply by guiccbr** [reply]
@joshuayoes , just to let you know that I opened a PR a few days ago. Thanks!

**Reply by joshuayoes** [reply]
Thanks for making that PR! I left a few review comments
